android/src/main/java/com/oney/WebRTCModule/GetUserMediaImpl.java

```diff
@@ -241,9 +227,137 @@ void disposeTrack(String id) {
 
     void switchCamera(String trackId) {
         TrackPrivate track = tracks.get(trackId);
-        if (track != null && track.videoCaptureController != null) {
-            track.videoCaptureController.switchCamera();
+        if (track != null && track.videoCaptureController instanceof CameraVideoCaptureController) {
+            CameraVideoCaptureController videoCaptureController = (CameraVideoCaptureController) track.videoCaptureController;
+            videoCaptureController.switchCamera();
+        }
+    }
+
+    public void getDisplayMedia(Promise promise) {
+        if (this.promise != null) {
+            promise.reject(new RuntimeException("Another operation is pending."));
+            return;
+        }
+
+        Activity currentActivity = this.reactContext.getCurrentActivity();
+        if (currentActivity == null) {
+            promise.reject(new RuntimeException("No current Activity."));
+            return;
+        }
+
+        this.promise = promise;
```

Comment:
Can we call this something less generic than "promise" so we know it's related to getDisplayMedia please?